openeo/rest/userfile.py

```diff
@@ -0,0 +1,58 @@
+import typing
+from typing import Any, Dict, List, Union
+from pathlib import Path
+
+if typing.TYPE_CHECKING:
+    # Imports for type checking only (circular import issue at runtime).
+    from openeo.rest.connection import Connection
+
+
+class UserFile:
+    """Represents a file in the user-workspace of openeo."""
+
+    def __init__(self, path: str, connection: 'Connection', metadata: Dict[str, Any] = {}):
+        self.path = path
+        self.metadata = metadata
+        self.connection = connection
+
+    def __repr__(self):
+        return '<{c} file={i!r}>'.format(c=self.__class__.__name__, i=self.path)
+
+    def _get_endpoint(self) -> str:
+        return "/files/{}".format(self.path)
+
+    def get_metadata(self, key) -> Any:
+        """ Get metadata about the file, e.g. file size (key: 'size') or modification date (key: 'modified')."""
+        if key in self.metadata:
+            return self.metadata[key]
+        else:
+            return None
+
+    def download_file(self, target: Union[Path, str]) -> Path:
+        """ Downloads a user-uploaded file."""
+        # GET /files/{path}
+        response = self.connection.get(self._get_endpoint(), expected_status=200, stream=True)
+
+        path = Path(target)
```

Comment:
I think it would be handy if the user can also just specify a folder as target ("." by default even), and the client just takes the remote file's name/path

for example see https://github.com/Open-EO/openeo-python-client/blob/d58650c4c39e486fda01252bece5ddc734a23f9e/openeo/rest/job.py#L307-L320

Reply:
Done